src/results/aggregator.rs

```diff
@@ -1,186 +1,111 @@
-/// * `upstream_search_engines` - Accepts a vector of search engine names which was selected by the
-/// * `request_timeout` - Accepts a time (secs) as a value which controls the server request timeout.
-/// user through the UI or the config file.
-///
-/// # Error
-///
-/// Returns an error a reqwest and scraping selector errors if any error occurs in the results
-/// function in either `searx` or `duckduckgo` or both otherwise returns a `SearchResults struct`
-/// containing appropriate values.
-pub async fn aggregate(
-    query: &str,
-    page: u32,
-    random_delay: bool,
-    debug: bool,
-    upstream_search_engines: &[EngineHandler],
-    request_timeout: u8,
-    safe_search: u8,
-) -> Result<SearchResults, Box<dyn std::error::Error>> {
-    let client = CLIENT.get_or_init(|| {
-        ClientBuilder::new()
-            .timeout(Duration::from_secs(request_timeout as u64)) // Add timeout to request to avoid DDOSing the server
-            .https_only(true)
-            .gzip(true)
-            .brotli(true)
-            .build()
-            .unwrap()
-    });
-
-    let user_agent: &str = random_user_agent();
-
-    // Add a random delay before making the request.
-    if random_delay || !debug {
-        let nanos = SystemTime::now().duration_since(UNIX_EPOCH)?.subsec_nanos() as f32;
-        let delay = ((nanos / 1_0000_0000 as f32).floor() as u64) + 1;
-        tokio::time::sleep(Duration::from_secs(delay)).await;
-    }
-
-    let mut names: Vec<&str> = Vec::with_capacity(0);
-
-    // create tasks for upstream result fetching
-    let mut tasks: FutureVec = FutureVec::new();
-
-    for engine_handler in upstream_search_engines {
-        let (name, search_engine) = engine_handler.to_owned().into_name_engine();
-        names.push(name);
-        let query: String = query.to_owned();
-        tasks.push(tokio::spawn(async move {
-            search_engine
-                .results(&query, page, user_agent, client, safe_search)
-                .await
-        }));
-    }
-
-    // get upstream responses
-    let mut responses = Vec::with_capacity(tasks.len());
-
-    for task in tasks {
-        if let Ok(result) = task.await {
-            responses.push(result)
-        }
-    }
-
-    // aggregate search results, removing duplicates and handling errors the upstream engines returned
-    let mut result_map: HashMap<String, SearchResult> = HashMap::new();
-    let mut engine_errors_info: Vec<EngineErrorInfo> = Vec::new();
-
-    let mut handle_error = |error: &Report<EngineError>, engine_name: &'static str| {
-        log::error!("Engine Error: {:?}", error);
-        engine_errors_info.push(EngineErrorInfo::new(
-            error.downcast_ref::<EngineError>().unwrap(),
-            engine_name,
-        ));
-    };
-
-    for _ in 0..responses.len() {
-        let response = responses.pop().unwrap();
-        let engine = names.pop().unwrap();
+pub struct Ranker;
+
+impl Ranker {
```

Comment:
```suggestion
pub struct Aggregator;

impl Aggregrator {
```

I would suggest renaming this struct to Aggregator, as we feel having the name Ranker would be very confusing for anybody working on the project. As there is currently a work for having a ranking algorithm being worked on which can cause confusion between the two. I hope you got my point :slightly_smiling_face: